pkg/agent/controller/networkpolicy/networkpolicy_controller.go

```diff
@@ -100,6 +100,56 @@ func (c *Controller) GetAppliedToGroupNum() int {
 	return c.ruleCache.GetAppliedToGroupNum()
 }
 
+func (c *Controller) GetRules() []v1beta1.NetworkPolicyRule {
+	c.ruleCache.policySetLock.RLock()
+	defer c.ruleCache.policySetLock.RUnlock()
+	var ret []v1beta1.NetworkPolicyRule
+	for uid := range c.ruleCache.policySet {
+		rules, _ := c.ruleCache.rules.ByIndex(policyIndex, uid)
+		for _, ruleObj := range rules {
+			rule := ruleObj.(*rule)
+			ret = append(ret, v1beta1.NetworkPolicyRule{
+				Direction: rule.Direction,
+				From:      rule.From,
+				To:        rule.To,
+				Services:  rule.Services,
+			})
```

Comment:
I was expecting the `rule` struct because it could contain all elements to represent a policy rule including appliedToGroups and policy UID, I guess you use `v1beta1.NetworkPolicyRule` is to leverage its serialization tag, then maybe keep your original implementation because the appliedToGroup and policy identifier are still necessary.

Reply:
Sure, updated.